includes/collection/class-actors.php

```diff
@@ -76,6 +77,14 @@ public static function get_by_id( $user_id ) {
 	 * @return User|Blog|Application|WP_Error The Actor or WP_Error if user not found.
 	 */
 	public static function get_by_username( $username ) {
+		$old_domain = \get_option( 'activitypub_old_domain' );
+
+		// Special case for Blog Actor on old domain.
+		if ( $old_domain && $old_domain === $username && Http::is_domain_match( $old_domain ) ) {
+			// Return a new Blog instance which will load the cached data in its constructor.
+			return new Blog();
+		}
```

Comment:
I would prefer to use filters for that! we should not (hard)code the functionality in too many places!

Reply:
it makes things way more complex, because the integration seems to be everywhere. through a filter it is easier to have all integrations in one class and it is unobtrusive. you can easily unhook the feature without doing too much changes in every possible class.

Reply:
I think you need this code maybe once in the livetime of your website, so it would be nice to even load the whole migration integration only if `$domain_moves_enabled` is enabled or `'activitypub_old_domain'` is set or based on some other indicator.